### PR TITLE
Day 2 queries

### PR DESCRIPTION
**Pull request by Ochiengsteven**
🦁 **Feature Implementation: Database Schema Modifications and Advanced SQL Queries**

This pull request encompasses several significant updates to our database schema and SQL queries, enhancing the functionality and data within the `animals` table.

**Database Schema Modifications:**

1. Added a new column `species` of type string to the `animals` table, modifying the `schema.sql` file.

**Data Insertion:**

2. Inserted new animals data into the `animals` table as follows:
   - Charmander: Born on Feb 8th, 2020, weighs -11kg, not neutered, never tried to escape.
   - Plantmon: Born on Nov 15th, 2021, weighs -5.7kg, neutered, tried to escape 2 times.
   - Squirtle: Born on Apr 2nd, 1993, weighs -12.13kg, not neutered, tried to escape 3 times.
   - Angemon: Born on Jun 12th, 2005, weighs -45kg, neutered, tried to escape once.
   - Boarmon: Born on Jun 7th, 2005, weighs 20.4kg, neutered, tried to escape 7 times.
   - Blossom: Born on Oct 13th, 1998, weighs 17kg, neutered, tried to escape 3 times.
   - Ditto: Born on May 14th, 2022, weighs 22kg, neutered, tried to escape 4 times.

**Transaction Management:**

3. Inside a transaction, updated the `species` column for all animals to "unspecified" and verified the change. Then rolled back the change and confirmed the species returned to its previous state.

4. Inside a transaction:
   - Updated the `species` column to "digimon" for animals with names ending in "mon."
   - Updated the `species` column to "pokemon" for animals with no species set.
   - Committed the transaction and verified that the changes persisted.

5. Inside a transaction, deleted all records in the `animals` table, then rolled back the transaction. Verified that all records remained intact.

6. Inside a transaction:
   - Deleted all animals born after Jan 1st, 2022.
   - Created a savepoint for the transaction.
   - Updated all animals' weights to be their weight multiplied by -1.
   - Rolled back to the savepoint.
   - Updated all animals' weights that are negative to be their weight multiplied by -1.
   - Committed the transaction.

**SQL Queries:**

7. Wrote SQL queries to answer various questions, including:
   - How many animals are there?
   - How many animals have never tried to escape?
   - What is the average weight of animals?
   - Who escapes the most, neutered or not neutered animals?
   - What is the minimum and maximum weight of each type of animal?
   - What is the average number of escape attempts per animal type for those born between 1990 and 2000?

**Screenshots:**

Attached screenshots with the results of SQL queries and transactions to this pull request description.

📸 **Screenshots with Query Results:**
screenshot1

screenshot2
screenshot3
screenshot4

### **SQL queries answers screenshots:**
**How many animals are there?**
total animals
**How many animals have never tried to escape?**
no escape
**What is the average weight of animals?**
avg weight
**Who escapes the most, neutered or not neutered animals?**
most escapes
**What is the minimum and maximum weight of each type of animal?**
min-max weight
**What is the average number of escape attempts per animal type for those born between 1990 and 2000?**
avarage escapes

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Ochiengsteven, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the future features, kindly add it as it is required.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: queries.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+COMMIT;
+
+-- Verify changes persist after commit
+SELECT * FROM animals;
+
+-- Transaction 3: Delete all records, then rollback and verify
+START TRANSACTION;
+
+DELETE FROM animals;
+
```
Comment (TuyishimireEric) [inline queries.sql:32]: - [ ] Good job so far 👏 👏 however, in this project we were required to write queries to answer the following questions:
    - How many animals are there?
    - How many animals have never tried to escape?
    - What is the average weight of animals?
    - Who escapes the most, neutered or not neutered animals?
    - What is the minimum and maximum weight of each type of animal?
    - What is the average number of escape attempts per animal type of those born between 1990 and 2000?


and also these queries should be added to your `queries.sql` file and to add screenshots with the results of your queries 👍 to complete the project requirements

**Review by TuyishimireEric — APPROVED** [review]
# Status: REQUIRED CHANGES :recycle: 
### Hi  @Ochiengsteven  👋 👋 
#### I'm @TuyishimireEric, Your code reviewer,
Good job so far 👍 
 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 
✔ Professional PR
✔ correct sql commands

## Required Changes ♻️

- Check the comments below 👇

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Ochiengsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good SQL Query design :white_check_mark: 
- [x] Pull request has a descriptive summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**